### PR TITLE
Add Scan QR code from clipboard

### LINKED ISSUE #2: Scan QR code from clipboard

### PR DESCRIPTION
**Pull request by raffifu**
this PR close #2 

The idea is to use [paste event](https://developer.mozilla.org/en-US/docs/Web/API/Element/paste_event) and then grab the files object from `event.clipboardData`

```
[praesentia.js]
....
const target = document.querySelector('body');

target.addEventListener('paste', (event) => {
    let paste = event.clipboardData || window.clipboardData;

    if(paste.files.length > 0) {
        html5QrcodeScanner.html5Qrcode.scanFile(paste.files[0], true)
        .then(decodedText => {
            processingDecodedText(decodedText)
        })
        .catch(err => {
            swal.fire('Error', 'Cannot read QR Code, Please make sure paste only the QR Code', 'error');
        });
    }
});
....
```

When the file object exist, call method `scanFile` from class `html5Qrcode`. it will return `decodedText` if successful and when fail will call `swal.fire`

### CAUTION
`html5Qrcode` instance is private field from `html5QrcodeScanner` see [this file on line (87)](https://github.com/mebjas/html5-qrcode/blob/bc2eb18d4d08226b4ea48a7403ca5e4590369ecb/src/html5-qrcode-scanner.ts#L76). the problem can be solved by creating `html5Qrcode` directly (use [Pro Mode](https://github.com/mebjas/html5-qrcode#pro-mode---if-you-want-to-implement-your-own-user-interface) as it says on the documentation)

### For demo visit this [link](https://drive.google.com/uc?id=1t_Wm5PlJA7CkWagMAn9Qj-jR_D85Fnff)

### CODE REVIEW

File: praesentia/static/praesentia.js [hunk 1]
```diff
@@ -6,10 +6,30 @@ let html5QrcodeScanner = new Html5QrcodeScanner(
...
+target.addEventListener('paste', (event) => {
+    let paste = event.clipboardData || window.clipboardData;
+
+    if(paste.files.length > 0) {
+        html5QrcodeScanner.html5Qrcode.scanFile(paste.files[0], true)
+        .then(decodedText => {
+            processingDecodedText(decodedText)
+        })
+        .catch(err => {
+            swal.fire('Error', 'Cannot read QR Code, Please make sure paste only the QR Code', 'error');
```
Comment (lc-at) [inline praesentia/static/praesentia.js]:
```suggestion
            swal.fire('Error', 'Cannot read QR code. Please make sure to only paste a QR code image.', 'error');
```

File: praesentia/static/praesentia.js [hunk 2]
```diff
@@ -6,10 +6,30 @@ let html5QrcodeScanner = new Html5QrcodeScanner(
         qrbox: 250,
     });
 
-
 html5QrcodeScanner.render(onScanSuccess);
 
+const target = document.querySelector('body');
+
+target.addEventListener('paste', (event) => {
```
Comment (lc-at) [inline praesentia/static/praesentia.js]: I have been using jQuery throughout the file. I don't know whether or not to continue using jQuery. Or, for the sake of consistency, should we just use jQuery?

File: praesentia/static/praesentia.js [hunk 3]
```diff
@@ -6,10 +6,30 @@ let html5QrcodeScanner = new Html5QrcodeScanner(
...
+        });
+    }
+});
+
+
 function onScanSuccess(decodedText, _) {
+    processingDecodedText(decodedText)
+}
+
+function processingDecodedText(decodedText) {
```
Comment (lc-at) [inline praesentia/static/praesentia.js]:
```suggestion
function processDecodedText(decodedText) {
```

File: praesentia/static/praesentia.js [hunk 4]
```diff
@@ -6,10 +6,30 @@ let html5QrcodeScanner = new Html5QrcodeScanner(
...
+        })
+        .catch(err => {
+            swal.fire('Error', 'Cannot read QR code. Please make sure to only paste a QR code image.', 'error');
+        });
+    }
+});
+
+
 function onScanSuccess(decodedText, _) {
+    processingDecodedText(decodedText)
```
Comment (lc-at) [inline praesentia/static/praesentia.js]:
```suggestion
    processDecodedText(decodedText)
```

**Comment by lc-at** [comment]
Thank you for your PR. I really appreciate that.

Apologies, I messed up with the review. Haven't reviewed a pull request. I believe this is my first time. This PR is ready to merge, but I am still unsure about whether or not to use jQuery. I think it should be consistent. What do you think?

File: praesentia/static/praesentia.js [hunk 5]
```diff
@@ -6,10 +6,30 @@ let html5QrcodeScanner = new Html5QrcodeScanner(
...
 
+const target = document.querySelector('body');
+
+target.addEventListener('paste', (event) => {
+    let paste = event.clipboardData || window.clipboardData;
+
+    if(paste.files.length > 0) {
+        html5QrcodeScanner.html5Qrcode.scanFile(paste.files[0], true)
+        .then(decodedText => {
+            processingDecodedText(decodedText)
```
Comment (lc-at) [inline praesentia/static/praesentia.js]:
```suggestion
            processDecodedText(decodedText)
```

File: praesentia/static/praesentia.js (hunk 2 above)
Comment (raffifu) [reply praesentia/static/praesentia.js]: I guess so, for consistency of this project. I can translate my code using jQuery

**Comment by raffifu** [comment]
I think so, this project can be developed further. So, the code must be consistent. I will fix that in the next commit

**Comment by lc-at** [comment]
Thanks. I will merge after that commit.

**Comment by raffifu** [comment]
74a669f94fb1a24d85a8d916cabb137ba446fa5c Now this PR follow jQuery format

```
[praesentia.js]
....
$('body').on('paste', event => {
    let paste = event.originalEvent.clipboardData || window.clipboardData;

    if(paste.files.length > 0) {
        html5QrcodeScanner.html5Qrcode.scanFile(paste.files[0], true)
        .then(decodedText => {
            processDecodedText(decodedText)
        })
        .catch(err => {
            swal.fire('Error', 'Cannot read QR code. Please make sure to only paste a QR code image.', 'error');
        });
    }

    event.preventDefault()
});
....
```

I also added `event.preventDefault()`  as said on[ this page](https://developer.mozilla.org/en-US/docs/Web/API/Element/paste_event)

**Comment by lc-at** [comment]
I have just tested it and it worked really well just as expected. Before merging, let me just add a little label to the interface, so users can notice this feature.

**Comment by lc-at** [comment]
Fixes: #2

**Comment by lc-at** [comment]
Thanks, @raffifu.

**Pull request merged**